api/src/main/java/org/openmrs/module/addresshierarchy/config/AddressConfigurationLoader.java

```diff
@@ -0,0 +1,274 @@
+package org.openmrs.module.addresshierarchy.config;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.List;
+
+import org.apache.commons.io.FileUtils;
+import org.apache.commons.io.IOUtils;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.openmrs.GlobalProperty;
+import org.openmrs.api.AdministrationService;
+import org.openmrs.api.context.Context;
+import org.openmrs.module.addresshierarchy.AddressHierarchyLevel;
+import org.openmrs.module.addresshierarchy.service.AddressHierarchyService;
+import org.openmrs.module.addresshierarchy.util.AddressHierarchyImportUtil;
+import org.openmrs.util.OpenmrsConstants;
+import org.openmrs.util.OpenmrsUtil;
+
+import com.thoughtworks.xstream.XStream;
+import com.thoughtworks.xstream.io.xml.DomDriver;
+
+/**
+ * Responsible for loading the address configuration appropriately
+ */
+public class AddressConfigurationLoader {
+
+	protected static final String ADDR_CONFIG_FILE_NAME = "addressConfiguration.xml";
+
+	public static final String NOT_COMPUTABLE_CHECKSUM = "not_computable_checksum";
+	public static final String NOT_READABLE_CHECKSUM = "not_readadble_checksum";
+
+	public static final String CHECKSUM_FILE_EXT = "checksum";
+
+	protected static Log log = LogFactory.getLog(AddressConfigurationLoader.class);
+
+	/**
+	 * @return The path to the configuration subdirectory.
+	 */
+	public static String getSubdirConfigPath() {
+		return new StringBuilder()
+				.append(OpenmrsUtil.getApplicationDataDirectory())
+				.append("configuration").append(File.separator).append("addresshierarchy")
+				.toString();
+	}
+
+	public static void loadAddressConfiguration() {
+
+		final ConfigLoaderUtil configUtil = new ConfigLoaderUtil(getSubdirConfigPath());
+
+		String xmlConfigFileName = ADDR_CONFIG_FILE_NAME;
+
+		File configFile = configUtil.getFile(xmlConfigFileName);
+		if (!configFile.exists()) {
+			log.warn("Address hierarchy configuration file appears invalid, skipping the loading process: " + xmlConfigFileName);
```

Comment:
Yeah, I understand that trade-off.  I suppose it could be another configuration option, as to whether to fail hard or warn soft upon failure to load the address hierarchy.  Regarding the level of logging, I'm happy to let you choose, I just wanted to point out that we should avoid logging warnings when the user is not in fact doing anything wrong and no warning state exists.